pkg/cluster/cluster.go

```diff
@@ -34,6 +36,18 @@ func (c *Cluster) Start(ctx context.Context) (<-chan struct{}, error) {
 		return nil, errors.Wrap(err, "init cluster datastore and https")
 	}
 
+	if c.config.DisableETCD {
+		ready := make(chan struct{})
+		defer close(ready)
+		etcdAddress, err := url.Parse(c.config.JoinURL)
+		etcdProxy, err := etcd.NewETCDProxy(true, c.config.DataDir, "https://"+etcdAddress.Hostname()+":2379")
```

Comment:
What's the benefit of setting up a tunnel for this? There's some complicated logic in the etcd client libraries that it feels like we're bypassing by hiding the whole cluster behind a single proxy endpoint.

What does it give us over just passing Kubernetes the actual etcd endpoint client addresses and just letting core etcd client handle retries and faillover? Is it just the ability to add/remove cluster members at runtime without having to restart the apiserver?

Reply:
yes exactly it mitigate the need for restarting the api everytime we add new etcd only node